pkg/fanal/artifact/image/image.go

```diff
@@ -173,48 +172,33 @@ func (a Artifact) Clean(_ artifact.Reference) error {
 }
 
 // findMatchingReference finds a RepoTag or RepoDigest that matches the artifact name
-func (a Artifact) findMatchingReference(artifactName string, repoTags, repoDigests []string) image.Reference {
+func (a Artifact) findMatchingReference(artifactName string, repoTags, repoDigests []string) name.Reference {
 	// Convert strings to typed references
-	parsedTags := a.parseRepoTags(repoTags)
-	parsedDigests := a.parseRepoDigests(repoDigests)
-
-	ref := a.findMatchingRepoReference(artifactName, parsedTags, parsedDigests)
-	return image.NewReference(ref)
+	references := a.parseImageReferences(slices.Concat(repoTags, repoDigests))
+	return a.findMatchingRepoReference(artifactName, references)
 }
 
-// parseRepoTags parses repo tags into name.Tag
-func (a Artifact) parseRepoTags(repoTags []string) []name.Tag {
-	return lo.FilterMap(repoTags, func(tagStr string, _ int) (name.Tag, bool) {
-		tag, err := name.NewTag(tagStr)
+// parseImageReferences parses repo tags/digests into name.Reference
+func (a Artifact) parseImageReferences(references []string) []name.Reference {
+	return lo.FilterMap(references, func(ref string, _ int) (name.Reference, bool) {
+		tag, err := name.ParseReference(ref)
 		if err != nil {
-			a.logger.Debug("Failed to parse repo tag", log.String("tag", tagStr), log.Err(err))
-			return name.Tag{}, false
+			a.logger.Debug("Failed to parse repo tag", log.String("ref", ref), log.Err(err))
```

Comment:
```suggestion
			a.logger.Debug("Failed to parse repo tag/digest", log.String("ref", ref), log.Err(err))
```
or something like this